### PR TITLE
ER-1377_show link to restore access for blocked provider.

### PR DESCRIPTION
**Pull request by shomavg**

### CODE REVIEW

File: src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            var vm = await _orchestrator.GetBlockedOrganisationViewModel(ukprn);
+            return View(vm);
+        }
+
+        [HttpPost("", Name = RouteNames.UnBlockProvider_Confirm_Post)]
+        public async Task<IActionResult> RestoreAccess(UnBlockTrainingProviderEditModel model)
+        {
+            if (ModelState.IsValid == false)
+            {
+                return View(new UnBlockTrainingProviderEditModel() { Ukprn = model.Ukprn, ProviderName = model.ProviderName });
```
Comment (chrisjensenuk) [inline src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs]: Naming is a bit wonky between action names, view models and views. Try and keep them aligned if possible.

Can we change usages of the word `RestoreAccess` to `Unblock`.  The front-end can say 'Restore Access' but I think in the code keeping to block/unblock is easier to understand. 

Change the ViewModel from `UnBlockTrainingProviderEditModel` to `ConfirmTrainingProviderUnblockingViewModel` so it aligns with the existing `ConfirmTrainingProviderBlockingViewModel`.

Change this Function name to `ConfirmTrainingProviderBlocking`

Change the view from `RestoreAccess.cshtml` to `ConfirmTrainingProviderUnblocking.cshtml`.

...etc

File: src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                await _orchestrator.UnBlockProviderAsync(model.Ukprn, User.GetVacancyUser());
+                return RedirectToRoute(RouteNames.UnBlockProvider_Acknowledgement_Get);
+            }
+            return RedirectToRoute(RouteNames.BlockedOrganisations_Get);
+        }
+
+        [HttpGet("unblocking-acknowledgement", Name = RouteNames.UnBlockProvider_Acknowledgement_Get)]
+        public async Task<IActionResult> ProviderUnBlockedAcknowledgement()
+        {
+            var data = TempData.Peek(TempDataKeys.UnBlockedProviderUkprnKey)?.ToString();
```
Comment (chrisjensenuk) [inline src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs]: Do we need to peek if we are going to remove it? Maybe just read it.

File: src/QA/QA.Web/Orchestrators/UnBlockOrganisationOrchestrator.cs [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+            return blockedOrganisation?.BlockedStatus == BlockedStatus.Blocked;
+        }
+
+        public Task UnBlockProviderAsync(long ukprn, VacancyUser user)
+        {
+            var command = new UnBlockProviderCommand(ukprn, user, _timeProvider.Now);
+            return _messaging.SendCommandAsync(command);
+        }
+
+        public async Task<UnBlockTrainingProviderEditModel> GetBlockedOrganisationViewModel(long ukprn)
```
Comment (chrisjensenuk) [inline src/QA/QA.Web/Orchestrators/UnBlockOrganisationOrchestrator.cs]: Pedantic but should `UnBlock` be `Unblock`?

Also `GetBlockedOrganisationViewModel` is not returning what it says it is.  See my comment about naming above.

File: src/QA/QA.Web/Configuration/Routing/RoutePaths.cs [hunk 4]
```diff
@@ -7,6 +7,7 @@ public static class RoutePaths
         public const string ExceptionHandlingPath = "/error/handle";
         public const string BlockedOrganisations = "/blockedorganisations";
         public const string WithdrawVacancyPath = "/withdraw";
+        public const string UnBlockOrganisationRoutePath = "/restoreaccess";
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/QA/QA.Web/Configuration/Routing/RoutePaths.cs]: 👍

File: src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs (hunk 1 above)
Comment (shomavg) [reply src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs]: 👍

File: src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        {
+            if (ModelState.IsValid == false)
+            {
+                return View(new UnBlockTrainingProviderEditModel() { Ukprn = model.Ukprn, ProviderName = model.ProviderName });
+            }
+            TempData[TempDataKeys.UnBlockedProviderUkprnKey] = model.Ukprn;
+            var isBlocked = await _orchestrator.IsProviderAlreadyBlocked(model.Ukprn);
+            if (!isBlocked)
+            {
+                return View(new UnBlockTrainingProviderEditModel() { Ukprn = model.Ukprn, ProviderName = model.ProviderName });
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs]: 👍

File: src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+            }
+
+            if (model.CanRestoreAccess)
+            {
+                await _orchestrator.UnBlockProviderAsync(model.Ukprn, User.GetVacancyUser());
```
[the remark this answers is not in the document]
Comment (shomavg) [reply src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs]: 👍

File: src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs (hunk 2 above)
Comment (shomavg) [reply src/QA/QA.Web/Controllers/UnBlockOrganisationController.cs]: 👍

File: src/QA/QA.Web/Orchestrators/UnBlockOrganisationOrchestrator.cs (hunk 3 above)
Comment (shomavg) [reply src/QA/QA.Web/Orchestrators/UnBlockOrganisationOrchestrator.cs]: 👍

**Pull request merged**